sys/ztimer/Makefile.dep

```diff
@@ -38,6 +38,12 @@ ifneq (,$(filter xtimer_on_ztimer,$(USEMODULE)))
   PSEUDOMODULES += xtimer_on_ztimer
 endif
 
+# make evtimer use ztimer_msec as low level timer
+ifneq (,$(filter evtimer_on_ztimer,$(USEMODULE)))
+  USEMODULE += ztimer_msec
+  PSEUDOMODULES += evtimer_on_ztimer
```

Comment:
This should be moved to `makefiles/pseudomodules.mk`

Reply:
Ok. Should I also move the `xtimer_on_ztimer` pseudo module to that file?
Otherwise this isn't consistent ;)

Reply:
Yes, this should certainly be fixed. It is currently the only place in RIOT `PSEUDOMODULES` is touched within a `Makefile.dep` and I really would avoid this pattern from sneaking in. The problem with those `Makefile.dep`s is that they are recursively included again and again until the content in `USEMODULE` (after sorting and de-duplicating) converges. And e.g. for `make info-boards-supported` the dependency resolution is done for each board RIOT supports. Thus, the `Makefile.dep`s should be as slim as possible.

But I would like to see this in a separate PR. No need to delay this PR for that. Do you want to create this PR? (If so, you could cherry-pick the indent fix also into that. It would be nice to also extend `dist/tools/buildsystem_sanity_check/check.sh` to enforce that the pattern `PSEUDOMODULE[\t ]*[+:]*=` is never found in any `Makefile.dep`. I would personally also would like to relocate those in the `Makefile.include` for consistency, but that is not so pressing as those in the `Makefile.dep`.)

Reply:
(Please also state if you prefer not to open the PR. I would create the PR in that case.)

Reply:
Thanks for the detailed explanation!
I'll open a separate PR for that.

Reply:
Done